packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -0,0 +1,22 @@
+import { declare } from "@babel/helper-plugin-utils";
+import type { types as t } from "@babel/core";
+import type { NodePath } from "@babel/traverse";
+
+export default declare(function () {
+  return {
+    visitor: {
+      "ImportDeclaration|ExportAllDeclaration|ExportNamedDeclaration"(
+        path: NodePath<
+          | t.ImportDeclaration
+          | t.ExportAllDeclaration
+          | t.ExportNamedDeclaration
+        >,
+      ) {
+        const { source } = path.node;
```

Comment:
We can also early return if `importKind` is `type`, since they will be stripped by the typescript transform anyway.